runtime/src/crowdfund.rs

```diff
@@ -0,0 +1,955 @@
+				&Self::fund_account_id(index),
+				balance,
+				WithdrawReason::Transfer,
+				ExistenceRequirement::AllowDeath
+			)?);
+
+			Self::contribution_kill(index, &who);
+			fund.raised = fund.raised.saturating_sub(balance);
+
+			<Funds<T>>::insert(index, &fund);
+
+			Self::deposit_event(RawEvent::Withdrew(who, index, balance));
+		}
+		
+		/// Remove a fund after either: it was unsuccessful and it timed out; or it was successful
+		/// but it has been retired from its parachain slot. This places any deposits that were not
+		/// withdrawn into the treasury.
+		fn dissolve(origin, #[compact] index: FundIndex) {
+			let _ = ensure_signed(origin)?;
+
+			let fund = Self::funds(index).ok_or("invalid fund index")?;
+			ensure!(fund.parachain.is_none(), "cannot dissolve fund with active parachain");
+			let now = <system::Module<T>>::block_number();
+			ensure!(now >= fund.end + T::RetirementPeriod::get(), "retirement period not over");
+
+			let account = Self::fund_account_id(index);
+
+			// Avoid using transfer to ensure we don't pay any fees.
+			let _ = T::Currency::resolve_into_existing(&fund.owner, T::Currency::withdraw(
+				&account,
+				fund.deposit,
+				WithdrawReason::Transfer,
+				ExistenceRequirement::AllowDeath
+			)?);
+
+			T::OrphanedFunds::on_unbalanced(T::Currency::withdraw(
+				&account,
+				fund.raised,
+				WithdrawReason::Transfer,
+				ExistenceRequirement::AllowDeath
+			)?);
+
+			Self::crowdfund_kill(index);
+			<Funds<T>>::remove(index);
+
+			Self::deposit_event(RawEvent::Dissolved(index));
+		}
+		
+		fn on_finalize(n: T::BlockNumber) {
+			if <slots::Module<T>>::is_ending(n).is_some() {
+				for (fund, index) in NewRaise::take().into_iter().filter_map(|i| Self::funds(i).map(|f| (f, i)))
+				{
+					let bidder = slots::Bidder::New(slots::NewBidder {
+						who: Self::fund_account_id(index),
+						/// FundIndex and slots::SubId happen to be the same type (u32). If this
+						/// ever changes, then some sort of conversion will be needed here.
+						sub: index,
+					});
+
+					// Care needs to be taken by the crowdfund creator that this function will succeed given
+					// the crowdfunding configuration. We do some checks ahead of time in crowdfund `create`.
+					let _ = <slots::Module<T>>::handle_bid(
+						bidder,
+						<slots::Module<T>>::auction_counter(),
+						fund.first_slot,
+						fund.last_slot,
+						fund.raised,
+					);
+				}
+			}
+		}
+	}
+}
+
+impl<T: Trait> Module<T> {
+	/// The account ID of the fund pot.
+	///
+	/// This actually does computation. If you need to keep using it, then make sure you cache the
+	/// value and only call this once.
+	pub fn fund_account_id(index: FundIndex) -> T::AccountId {
+		MODULE_ID.into_sub_account(index)
+	}
+
+	pub fn id_from_index(index: FundIndex) -> Vec<u8> {
+		let mut buf = Vec::new();
+		buf.extend_from_slice(b"crowdfund");
+		buf.extend_from_slice(&index.to_le_bytes()[..]);
+
+		CHILD_STORAGE_KEY_PREFIX.iter()
```

Comment:
```suggestion
		CHILD_STORAGE_KEY_PREFIX.into_iter()
```

Reply:
Where is this suggestion coming from? What is the difference?

Reply:
`.iter()` clones the object which isn't necessary because you're not using `CHILD_STORAGE_KEY_PREFIX` again so you might as well use `.into_iter()`, unless I'm missing something

Reply:
@shawntabrizi  For clarification, `into_iter` is preferrable in the scenario when you can consume the iterator because the object is not used thereafter. If there was an encoding requirement, `iter` would be preferred, but there is no compilation error so this is not the case.